### PR TITLE
Fix the resend verification email link for respondents that have changed email addresses

### PR DESCRIPTION
**Pull request by gavinleeedwards**
# Motivation and Context
Service desk call TASK0241375

A bug exists where if a respondent has requested a change of email address then requested the verification email to be resent, the link was not working and redirected to a 'URL not found' page.

# How to test?
- Register a new respondent without verifying the account and check that you can resend the verification email
- Change the respondent's email address
- Click the link to resend the verification email
- The link should now work and inform you that the email is being sent to their new address

### CODE REVIEW

File: response_operations_ui/views/reporting_units.py [hunk 1]
```diff
@@ -169,6 +169,8 @@ def search_reporting_units():
 def view_resend_verification(ru_ref, party_id):
     logger.debug("Re-send verification email requested", ru_ref=ru_ref, party_id=party_id)
     respondent = party_controller.get_respondent_by_party_id(party_id)
+    if 'pendingEmailAddress' in respondent:
+        return render_template('re-send-verification-email.html', ru_ref=ru_ref, email=respondent['pendingEmailAddress'])
```
Comment (gemmai95) [inline response_operations_ui/views/reporting_units.py]: Travis is complaining about this line being too long.

File: response_operations_ui/templates/reporting-unit.html [hunk 2]
```diff
@@ -148,7 +148,7 @@ <h4 class="neptune">Respondents</h4>
                 <dd>
                 Verification email sent
                 <br>
-                  <a href="{{ url_for('reporting_unit_bp.resend_verification', ru_ref=ru_ref, party_id=party_id) }}">Re-send verification email</a>
+                  <a href="{{ url_for('reporting_unit_bp.resend_verification', ru_ref=ru_ref, party_id=respondent.id) }}">Re-send verification email</a>
```
[the remark this answers is not in the document]
Comment (jcox-dev) [reply response_operations_ui/templates/reporting-unit.html]: @shahi645 looks like the kwarg should be `party_id`: [source](https://github.com/ONSdigital/response-operations-ui/blob/ee6776d3378586185308369ce92f91c0ed960130/response_operations_ui/views/reporting_units.py#L167).

[the remark this answers is not in the document]
Comment (shahi645) [reply response_operations_ui/templates/reporting-unit.html]: ahh ok, my bad, just looks a bit odd

Comment (tej99) [inline response_operations_ui/templates/reporting-unit.html]: reporting_unit_bp.view_resend_verification?

The user journey here should be when the user clicks on Re-send verification email they are directed to view resend verification page (this is a speed bump so the user does not accidentally click the resend button). 

You can follow the way it is being done [here](https://github.com/ONSdigital/response-operations-ui/blob/f055314591a205eaed8e92994ffe96e9b8d1d3ad/response_operations_ui/templates/reporting-unit.html#L184).

File: response_operations_ui/views/reporting_units.py [hunk 3]
```diff
@@ -169,7 +169,11 @@ def search_reporting_units():
 def view_resend_verification(ru_ref, party_id):
     logger.debug("Re-send verification email requested", ru_ref=ru_ref, party_id=party_id)
     respondent = party_controller.get_respondent_by_party_id(party_id)
-    return render_template('re-send-verification-email.html', ru_ref=ru_ref, email=respondent['emailAddress'])
+    if 'pendingEmailAddress' in respondent:
+        return render_template('re-send-verification-email.html', ru_ref=ru_ref,
```
Comment (tej99) [inline response_operations_ui/views/reporting_units.py]:
```
email = respondent['pendingEmailAddress'] if 'pendingEmailAddress' in respondent else respondent['emailAddress']

return render_template('re-send-verification-email.html', ru_ref=ru_ref, email=email)
```

I feel like its easier to read this way ^^

Comment (tej99) [reply response_operations_ui/views/reporting_units.py]: Also looks like you will always have a key 'pendingEmailAddress' in respondent. [source](https://github.com/ONSdigital/ras-party/blob/4e4b293c3f6e555998e174d50ed566f901985ef0/ras_party/models/models.py#L266)

Maybe 
`email = respondent['pendingEmailAddress'] if respondent['pendingEmailAddress'] else respondent['emailAddress'] ` ??

File: response_operations_ui/templates/reporting-unit.html (hunk 2 above)
Comment (gavinleeedwards) [reply response_operations_ui/templates/reporting-unit.html]: In what scenario are you seeing this?  For me, it always redirects back to the previous page as it should.

Comment (tej99) [reply response_operations_ui/templates/reporting-unit.html]: So it shouldn't be doing that (redirects back to the previous page). It should take you to a confirmation page which is rendered in the reporting_unit_bp.view_resend_verification function. Once you've confirmed it should redirect to the previous page.

File: response_operations_ui/views/reporting_units.py (hunk 3 above)
Comment (gavinleeedwards) [reply response_operations_ui/views/reporting_units.py]: Done 👍

**Pull request merged**